### DISCUSSION TITLE
Telerik.Web.UI and some issues

### DISCUSSION THREAD

**Discussion by thienvc** [dnnsoftware/Dnn.Platform#6219] [General] [answered]
As we know, there have been security bugs related to it over time, it's easy: remove it and you're done
However, there are many old modules based on it, which cannot be replaced immediately in a short time, but we cannot buy a license just to use it temporarily and then replace it.
After a while of getting tired, I decided to use an assembly editor to edit, remove everything (namespace) related to upload, editor, ... Except for the controls that the old DNN still depends on such as DnnCombo, RadGrid, ...
And it's great, it works, I want to share with someone about this method to temporarily feel secure in the process of moving all the old modules to the new platform that does not depend on Telerik

**Comment by mitchelsellers** [comment, marked as answer]
> 
> 
> As we know, there have been security bugs related to it over time, it's easy: remove it and you're done
> 
> However, there are many old modules based on it, which cannot be replaced immediately in a short time, but we cannot buy a license just to use it temporarily and then replace it.
> 
> After a while of getting tired, I decided to use an assembly editor to edit, remove everything (namespace) related to upload, editor, ... Except for the controls that the old DNN still depends on such as DnnCombo, RadGrid, ...
> 
> And it's great, it works, I want to share with someone about this method to temporarily feel secure in the process of moving all the old modules to the new platform that does not depend on Telerik

If you are modifying the Telerik DLL, that is VERY against the license terms and I would not distribute that to anyone.

**Reply by thienvc** [reply]
I understand that, I also do not distribute or encourage anyone to do that.

**Comment by Timo-Breumelhof** [comment]
I understand your issue (we had the same with some client modules).
But, "it's easy: remove it and you're done" is a bit of a simplification in my opinion.
There are community members that spent a lot of time to get these controls removed form the DNN Core a essential DNN Platform had the same issue you have now.
BTW, you can buy a commercial Telerik license (which was not an option for the core) and keep your modules running.
